### PR TITLE
loader: support remote modules in load()

### PR DESCRIPTION
**Pull request by cxuu**
Hackathon done right 😎

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -55,11 +56,26 @@ var (
...
 )
 
 func init() {
 	flag.Parse()
 	stdlog.SetFlags(stdlog.Lshortfile)
+	// If depsFile unset, and if $(pwd)/isopod.deps exists, update depsFile.
+	if *depsFile == "" {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("Failed to working dir: %v", err)
```
Comment (jonnylangefeld) [inline main.go:69]:
```suggestion
			log.Fatalf("Failed to get working dir: %v", err)
```

File: main.go [hunk 2]
```diff
@@ -55,11 +56,26 @@ var (
...
+		}
+		guess := filepath.Join(workingDir, dep.DepsFile)
```
Comment (jonnylangefeld) [inline main.go:71]: maybe `DefaultDepsFilePath`?

File: pkg/dep/dep.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+// with the module loader to support subsequent load() statements.
+func Load(entryfile string) error {
+	pkgs := starlark.StringDict{
+		"git_repository": NewGitRepoBuiltin(),
+	}
+	thread := &starlark.Thread{
+		Load: loader.NewModulesLoaderWithPredeclaredPkgs(filepath.Dir(entryfile), pkgs).Load,
+	}
+
+	bytes, err := ioutil.ReadFile(entryfile)
```
Comment (jonnylangefeld) [inline pkg/dep/dep.go:93]: I recommend to use `filepath.Abs()` for this string. That way both relative and absolute paths work.

File: pkg/dep/git.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+func Shellf(format string, a ...interface{}) (string, error) {
+	s := fmt.Sprintf(format, a...)
+	log.V(1).Infof("Executing shell command: %v\n", s)
+	bytes, err := exec.Command("sh", "-c", s).CombinedOutput()
+	log.V(1).Infof("Shell command `%s' finished:\n%s", s, string(bytes))
+	return string(bytes), err
+}
+
+// gitCloneBashScript is a bash script to fetch Git files of a repo at a given commit sha, to a given path.
+func gitCloneShellScript(outputDir, gitRemoteURL, gitCommitSHA string) string {
```
Comment (jonnylangefeld) [inline pkg/dep/git.go:130]: This function would have been a good use case for go templates.

Comment (jonnylangefeld) [reply pkg/dep/git.go:130]: Not a super big fan of having this as a shell script... It adds a non-implicit dependency for `git` and makes it a little less usable in docker containers. Would be great to use a go git client.
But I understand that it's complicated including authentication and all...

File: pkg/dep/dep.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+var (
+	// asserts *AbstractDependency implements starlark.HasAttrs interface.
+	_ starlark.HasAttrs = (*AbstractDependency)(nil)
+
+	// Workspace is the directory that stages all Isopod-managed remote modules.
+	Workspace = "/tmp/isopod-workspace"
+)
+
+// AbstractDependency contains the common impl of all KubernetesVendor.
```
Comment (jonnylangefeld) [inline pkg/dep/dep.go:41]: This doc string isn't super clear

Comment (cxuu) [reply pkg/dep/dep.go]: I added more details

File: pkg/dep/git.go (hunk 4 above)
Comment (cxuu) [reply pkg/dep/git.go]: I refactored this to use go templates.

Good point about the implicit dependency on `git`. Authentication is not impossible with `go-git` package but using `git` command allows Isopod to use existing auth creds, which I think is a huge UX improvement.

File: pkg/dep/dep.go (hunk 3 above)
Comment (cxuu) [reply pkg/dep/dep.go]: done

File: main.go (hunk 2 above)
Comment (cxuu) [reply main.go]: done

File: .gitignore [hunk 6]
```diff
@@ -1,4 +1,4 @@
-isopod*
+isopod
```
[the remark this answers is not in the document]
Comment (cxuu) [reply .gitignore:3]: I added more patterns in here and a commend explaining why

**Pull request merged**